src/main/java/com/box/sdk/BoxTransactionalAPIConnection.java

```diff
@@ -0,0 +1,130 @@
+package com.box.sdk;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+import com.eclipsesource.json.JsonObject;
+
+/**
+ * Represents an authenticated transactional connection to the Box API.
+ *
+ * <p>This class handles everything for transactional API that isn't already handled by BoxAPIConnection.</p>
+ */
+public class BoxTransactionalAPIConnection extends BoxAPIConnection {
+
+    private static final String SUBJECT_TOKEN_TYPE = "urn:ietf:params:oauth:token-type:access_token";
+    private static final String GRANT_TYPE = "urn:ietf:params:oauth:grant-type:token-exchange";
+
+    /**
+     * Constructs a new BoxTransactionalAPIConnection that authenticates with an access token.
+     * @param  accessToken a transactional auth access token.
+     */
+    public BoxTransactionalAPIConnection(String accessToken) {
+        super(accessToken);
+        super.setAutoRefresh(false);
+    }
+
+    /**
+     * Request a scoped transactional token.
+     * @param accessToken application access token.
+     * @param scope scope of transactional token.
+     * @return a BoxAPIConnection which can be used to perform transactional requests.
+     */
+    public static BoxAPIConnection getTransactionConnection(String accessToken, String scope) {
+        return BoxTransactionalAPIConnection.getTransactionConnection(accessToken, scope, null);
+    }
+
+    /**
+     * Request a scoped transactional token for a particular resource.
+     * @param accessToken application access token.
+     * @param scope scope of transactional token.
+     * @param resource resource transactional token has access to.
+     * @return a BoxAPIConnection which can be used to perform transactional requests.
+     */
+    public static BoxAPIConnection getTransactionConnection(String accessToken, String scope, String resource) {
+        BoxAPIConnection apiConnection = new BoxAPIConnection(accessToken);
+
+        URL url;
+        try {
+            url = new URL(apiConnection.getTokenURL());
+        } catch (MalformedURLException e) {
+            assert false : "An invalid token URL indicates a bug in the SDK.";
+            throw new RuntimeException("An invalid token URL indicates a bug in the SDK.", e);
+        }
+
+        String urlParameters;
+        try {
+            urlParameters = String.format("grant_type=%s&subject_token=%s&subject_token_type=%s&scope=%s", GRANT_TYPE,
+                    accessToken, SUBJECT_TOKEN_TYPE, URLEncoder.encode(scope, "UTF-8"));
```

Comment:
accessToken not encoded.  Maybe doesn't matter, because everything will fail if it's invalid, but encoding seems like good hygiene.

You could replace the String.format and try/catch by simply using QueryStringBuilder.  Might be cleaner.